src/wh_server.c

```diff
@@ -54,42 +54,15 @@ int wh_Server_Init(whServerContext* server, whServerConfig* config)
     }
 
     memset(server, 0, sizeof(*server));
+    server->crypto = config->crypto;
+    server->nvm = config->nvm;
```

Comment:
How should we handle a `NULL` NVM? Disallow? 

Right now, this could be NULL and thankfully it only returns failure to the client since the lower-level NVM handlers NULL check, but I absolutely can see this potentially resulting in a NULL dereference somewhere at some point if we don't properly enforce. 

Also, note that the tests still all pass with NULL `server->nvm` since they don't actually check any return values :(

Reply:
Returning WH_ERROR_BADARGS is probably the correct thing as this captures the problem of a dynamic allocation that fails as well.